docs/en/concepts-and-designs/event.md

```diff
@@ -0,0 +1,53 @@
+# Events
+
+SkyWalking already supports the three pillars of observability, namely logs, metrics, and traces.
+While in a production system in reality, there are many other events that may affect the performance of the system, such as upgrading, rebooting, chaos testing, etc.
+Although some of these events can reflect from the logs, there are also many other events that can not, hence SkyWalking provides a more native way to collect these events.
+This doc covers the design of how the SkyWalking collect events and what events look lik in SkyWalking.
+
+## How to Report Events
+
+SkyWalking backend supports three protocols to collect events, gRPC, HTTP, and Kafka. Any agent or CLI that implements one of these protocols can report events to the SkyWalking.
+Currently, the officially supported clients to report events are:
+
+- [ ] Java Agent Toolkit: Use the Java agent toolkit to report events from inside the applications.
+- [ ] SkyWalking CLI: Use the CLI to report events from the command line interface.
+- [ ] Kubernetes Event Exporter: Deploy an event exporter to refine and report Kubernetes events.
+
+## Event Definition
+
+An event contains the following fields. The definitions of event can be found at the [protocol repo](#TODO)
+
+### UUID
+
+Unique ID of the event. Because an event may span a long period of time, the UUID is necessary to associate the start time with the end time of the same event. 
+
+### Object
+
+The object that the event occurs on. In the concepts of SkyWalking, the object is typically service, service instance, etc.
+
+### Name
+
+The name of the event. For example, `Start`, `Stop`, `Crash`, `Reboot`, `Upgrade` etc.
+
+### Type
+
+The type of the event. This field is friendly for UI visualization, where events of type `Normal` is considered as normal operations,
+while `Error` is considered as unexpected operations, such as `Crash` events, therefore we can mark them with different colors to be easier identified.
+
+### Message
+
+The detail of the event that describes why this event happened. This should be a one-line message that briefly describes why the event is reported. Examples of a `Crash` event may be something like `Run out of memory`.
+It's NOT encouraged to include the detailed logs of this event, such as the exception stack trace.
```

Comment:
Are you asking for collaboration-obs-entity? Such as traceid, logid(s), etc.

Reply:
I'm not sure what collaboration-obs-entity means. I'm just saying there should be a way to allow events to have user data. Besides the use case I mentioned above, the use case I was imagining was when the event is of type `Upgrade`; you could add `from.version=1.2`/`to.version=1.3` in the user data somewhere. That way UI/CLI/etc that display the events can show additional information related to the event.

Reply:
I think `from.version=1.2/to.version=1.3` is reasonable in the message. @kezhenxu94 WDYT? I think the typical thing should not be included  is `exception stack`, which belongs to logs or trace/span.

Reply:
> I think `from.version=1.2/to.version=1.3` is reasonable in the message. @kezhenxu94 WDYT? I think the typical thing should not be included  is `exception stack`, which belongs to logs or trace/span.

Exactly. `from.version=1.2/to.version=1.3` or something like `Upgrade from 1.2 to 1.3` is valid in my mind.

Reply:
If you guys don't feel it's valid then disregard it. My thinking here was mainly to provide a way for static typing. I suppose you could always just encode something in JSON and put it in `message` if you need something more structural.

Reply:
The message structure is really a little complex. We could offer a field with JSON type to describe it as parameters, which provides a structure type, but we should provide recommended data format examples, for upgrading, OOM, kill, etc.

Reply:
I agree about the recommended data formats. I just don't see how you do that with something as simple as `Run out of memory`. That's good for a summary but IMO it would be nice to have the ability to attribute additional information. I don't mind if it's through some JSON field, meta-map, etc.

Reply:
> I just don't see how you do that with something as simple as `Run out of memory`. That's good for a summary but IMO it would be nice to have the ability to attribute additional information. I don't mind if it's through some JSON field, meta-map, etc.

Having a JSON / meta-map field sounds good to me, but because we don't have static typings for that field, users are free to report random text of that JSON / meta-map field.

I'm also wondering how will the structured field be used. There are many events (some may be unknown for now), whose JSON field properties are indeterministic.

If we want to provide a JSON field, we may need to list (recommended) properties of the JSON field for some common events, right?

Reply:
> If we want to provide a JSON field, we may need to list (recommended) properties of the JSON field for some common events, right?

Yes. We need to. But we could add this one by one, such as in my mind, an upgrade event could be
```
- type: Upgrade
- message: Upgrade from ${prev_version} to ${latest_version}.
- parameter: {"prev_version" : "1.2"; "latest_version" : "1.3" }
```

Reply:
The above would be my preferred solution as well.

Reply:
I added the `parameters` field